File: codingjs/countYZ.js
function countYZ(str: string): number {
  if (str.length === 0) return 0;
  if (str.toLowerCase() === 'y' || str.toLowerCase() === 'z') return 1;
  const target = str.substring(0, 2).toLowerCase()
  if (/[yz][^a-z]/.test(target)) {
    return 1 + countYZ(str.substring(2));
  }
  return 0 + countYZ(str.substring(1));
}

File: codingbat/countYZ.java
public int countYZ(String str) {
  int count = 0;
  for (int i = 0; i < str.length(); i++) {
    char target = Character.toLowerCase(str.charAt(i));
    if ((i == str.length() - 1) && (target == 'y' || target == 'z')) {
      count++;
      continue;
    } 
    if ((target == 'y' || target == 'z') && !Character.isLetter(str.charAt(i + 1))) {
      count++;
    }
  }
  return count;
}

